src/main.ts

```diff
@@ -962,3 +985,35 @@ const cleanUpProxies = async () => {
     kill(proc.pid)
   })
 }
+
+const generateUniqueFileName = async ({
+  directory,
+  prefix,
+  ext,
+}: {
+  directory: string
+  prefix: string
+  ext: string
+}): Promise<string> => {
+  const timestamp = new Date().toISOString().split('T')[0] ?? ''
+  const template = `${prefix ? `${prefix} - ` : ''}${timestamp}${ext}`
+
+  let uniqueFileName = template
+  let fileExists = await access(path.join(directory, uniqueFileName))
+    .then(() => true)
+    .catch(() => false)
```

Comment:
Is there a reason why we shouldn't use `fs.existsSync` here instead? It'd look semantically more appropriate as it returns a boolean value.

Reply:
I think what you want to do is use `await fs.writeFile(path, data, { flag: 'ax+' })` (see [flags](https://nodejs.org/api/fs.html#file-system-flags)) and repeat that with a new suffix until the write succeeds.

Reply:
IMO it's okay to use these methods in certain places (like a one-off operation, during initialization) that don't compromise the event loop.

@allansson's suggestion seems to handle both the check and the write at the same time so we can consider that approach as well.

If using `fs.access`, we could condering cleaning up the then/catch block if we move it to a helper function since it's been used in a few places.

```ts
const fileExists = await checkFileExsits(path.join(directory, uniqueFileName)
```

Reply:
I rewrote the function to use `writeFile`